sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -739,7 +614,7 @@ protected final void removeTemporaryFiles(Set<String> knownFiles,
      * Provides a coder for {@link FileBasedSink.FileResult}.
      */
     public final Coder<FileResult> getFileResultCoder() {
-      return FileResultCoder.of();
+      return SerializableCoder.of(FileResult.class);
```

Comment:
Why did you change this back? FileResultCoder was explicitly added, because SerializableCoder is somewhat broken in streaming.